vm/src/vm/vm_core.rs

```diff
@@ -1024,6 +1026,26 @@ impl VirtualMachine {
             Err(TraceError::TraceNotRelocated)
         }
     }
+
+    /// Returns a list of addresses of memory cells that constitute the public memory.
+    pub fn get_public_memory_addresses(&self) -> Result<Vec<(usize, &usize)>, MemoryError> {
+        if let Some(relocation_table) = &self.relocation_table {
+            Ok(self.segments.get_public_memory_addresses(relocation_table))
+        } else {
+            Err(MemoryError::UnrelocatedMemory)
+        }
+    }
+
+    pub fn get_memory_segment_addresses(&self) -> HashMap<&'static str, (usize, Option<usize>)> {
+        self.builtin_runners
+            .iter()
+            .map(|builtin| (builtin.name(), builtin.get_memory_segment_addresses()))
+            .collect()
+    }
+
+    pub fn get_trace(&self) -> &Option<Vec<TraceEntry>> {
+        &self.trace
+    }
 }
```

Comment:
Not needed, use `VirtualMachine::get_relocated_trace` instead

Reply:
done!